Homework/Duy_Truong/78_Subsets.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @param {number[]} nums
+ * @return {number[][]}
+ */
+var subsets = function(nums) {
+    let res = [[]];
+    for(let i=0;i<nums.length;i++) {
```

Comment:
this looks neat. Can you explain the solution?

Reply:
The idea is: I create an res array with empty array inside. and for each number in nums array, I try to get all the sub array created by the res array.
Ex: Input: nums = [1,2,3], res = [[]]
With i = 0 => size = res.length = 1. After loop from 0 to size(1) => res = [[], [1]]
With i = 1=> size = 2.... res = [[], [1], [2], [1,2]]...
The time complex is still O(N*2^N) and space complex is O(2^N) with N is nums length